scheduler/src/cook/compute_cluster.clj

```diff
@@ -343,17 +356,24 @@
    The location and state fields are special in this regard -- location can
    only transition from nil to non-nil, and state is validated in
    cluster-state-change-valid?."
-  [{current-location :location :as current-config}
-   {new-location :location :as new-config}]
-  (if (and (some? current-location)
-           (not= current-location new-location))
-    false
-    (let [dissoc-non-comparable-fields
-          #(-> %
-               (dissoc :state)
-               (dissoc :location))]
-      (= (dissoc-non-comparable-fields current-config)
-         (dissoc-non-comparable-fields new-config)))))
+  [{current-location :location current-features :features :as current-config}
+   {new-location :location new-features :features :as new-config}]
+  (cond (and (some? current-location)
+             (not= current-location new-location))
+        false
+
+        (and (seq current-features)
+             (not= current-features new-features))
```

Comment:
do we care about order when comparing? enough to call set on them first?

Reply:
Yeah, good call, will do

Reply:
Added a unit test and changed to call `set`